lib/private/Files/SetupManager.php

```diff
@@ -434,7 +435,10 @@ private function fullSetupRequired(IUser $user): bool {
 		// we perform a "cached" setup only after having done the full setup recently
 		// this is also used to trigger a full setup after handling events that are likely
 		// to change the available mounts
-		return !$this->cache->get($user->getUID());
+		if (!isset($this->fullSetupRequired[$user->getUID()])) {
+			$this->fullSetupRequired[$user->getUID()] = !$this->cache->get($user->getUID());
```

Comment:
```suggestion
			$this->fullSetupRequired[$user->getUID()] = $this->cache->get($user->getUID());
```

Isn't that inverted then on the return statement again?

Reply:
good catch, forgot to remove the invert on return